storage/service/src/reader_threads.rs

```diff
@@ -9,14 +9,28 @@
 //! based on these values.
 
 //---------------------------------------------------------------------------------------------------- Import
-use std::num::NonZeroUsize;
+use std::{num::NonZeroUsize, sync::Arc};
 
+use rayon::ThreadPool;
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+//---------------------------------------------------------------------------------------------------- init_thread_pool
+/// Initialize the reader thread-pool backed by `rayon`.
+pub fn init_thread_pool(reader_threads: ReaderThreads) -> Arc<ThreadPool> {
+    // How many reader threads to spawn?
+    let reader_count = reader_threads.as_threads().get();
+
+    Arc::new(
+        rayon::ThreadPoolBuilder::new()
+            .num_threads(reader_count)
+            .thread_name(|i| format!("{}::DatabaseReader({i})", module_path!()))
+            .build()
+            .unwrap(),
+    )
+}
+
 //---------------------------------------------------------------------------------------------------- ReaderThreads
-/// Amount of database reader threads to spawn when using [`service`](crate::service).
-///
 /// This controls how many reader thread `service`'s
 /// thread-pool will spawn to receive and send requests/responses.
```

Comment:
Leave `Amount of database reader threads to spawn`.

The following documentation mentioning a `service` module must be updated too.

Reply:
Ok, it does feel a bit redundant though as the next sentence says pretty much the same thing.